core/test/dataset_operations_test.cpp

```diff
@@ -680,3 +681,34 @@ TYPED_TEST(DatasetBinaryOpTest, dataset_lhs_dataproxy_rhs) {
     EXPECT_EQ(reference, item.data());
   }
 }
+
+Dataset non_trivial_2d_sparse() {
+  Dataset sparse;
+  auto var = makeVariable<double>({Dim::X, Dim::Y}, {3, Dimensions::Sparse});
+  var.sparseValues<double>()[0] = {1.5, 2.5, 3.5, 4.5, 5.5};
+  var.sparseValues<double>()[1] = {3.5, 4.5, 5.5, 6.5, 7.5};
+  var.sparseValues<double>()[2] = {1, 1, 2, 2, 2, 4, 4, 4, 6};
```

Comment:
Can you add some data the is less than the first bin bound, to check if it dropped as it should be.